managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedCursorImpl.java

```diff
@@ -480,7 +482,24 @@ public void operationComplete(ManagedCursorInfo info, Stat stat) {
                     // Need to proceed and read the last entry in the specified ledger to find out the last position
                     log.info("[{}] Consumer {} meta-data recover from ledger {}", ledger.getName(), name,
                             info.getCursorsLedgerId());
-                    recoverFromLedger(info, callback);
+                    recoverFromLedger(info, new VoidCallback() {
+                        @Override
+                        public void operationComplete() {
+                            LinkedHashSet<Long> ledgersDeletedButNotAcked = checkLedgersDeletedButNotAcked();
```

Comment:
Why do we need this checking logic? It looks like we already processed at cursor got `NonRecoverableLedgerException`

Reply:
these codes have been removed(This PR no longer call the checks when cursor is initializing)